File: hs/Mod_subtree.hs
module Mod_subtree ( subtree ) where
import Prelude hiding (and, or) 
import Debug.Trace
import Splib
import Types

subtree tree node = addChildren tree (newTreeOwn node) node

addChildren orig newTree node = aCHelper orig newTree node 0

aCHelper orig newTree node i 
   | i == length (children orig node) = newTree
   | otherwise = aCHelper orig (addChildren orig (addOwnLeaf newTree node (get (children orig node) i)) (get (children orig node) i)) node (i+1) 

File: cpp/Mod_subtree.cpp
#pragma once

Tree addChildren(Tree orig, Tree newTree, Node node){
	for(int i = 0; i<length(children(orig, node));i++){
		newTree = addOwnLeaf(newTree, node, get(children(orig, node), i));	
		newTree = addChildren(orig, newTree, get(children(orig, node), i));
	}
	
	return newTree;
}

Tree subtree(Tree tree, Node node){
	Tree newTree = newTreeOwn(node);
	newTree = addChildren(tree, newTree, node);
	return newTree;
}
